src/engine/guidance/post_processing.cpp

```diff
@@ -414,6 +414,14 @@ void collapseTurnAt(std::vector<RouteStep> &steps,
                      current_step.name_id == steps[two_back_index].name_id)
             {
                 steps[one_back_index].maneuver.instruction.type = TurnType::Continue;
+                // If we Merge onto the same street, we end up with a u-turn in some cases
+                if (bearingsAreReversed(util::bearing::reverseBearing(
+                                            one_back_step.intersections.front()
```

Comment:
@MoKob  to clarify:

the pattern currently is `var = default; if (condition) var = othervalue;` it is better to do `if (condition) var = othervalue else var = default;`